### PR TITLE
Changed to GroupBy

### LINKED ISSUE #59695: DOC: methods in see also section in the pandas.core.groupby.DataFrameGroupBy.agg are not hyperlinks
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

[pandas.core.groupby.DataFrameGroupBy.agg](https://pandas.pydata.org/docs/reference/api/pandas.core.groupby.DataFrameGroupBy.agg.html#pandas.core.groupby.DataFrameGroupBy.agg)

### Documentation problem

In the see also section the `DataFrame.groupby.apply`, `DataFrame.groupby.transform` and DataFrame..aggregate` methods though listed, they are not hyperlinks and thus the reader cannot navigate with ease but has to look for them instead 

### Suggested fix for documentation

I am not very familiar with the `sphinx` setup in `pandas` and thus cannot suggest an approach for fixing this. Perhaps there is a good reason that these methods are not hyperlinked, note sure

### PR DESCRIPTION
**Pull request by doshi-kevin**
- [ ] closes #59695  (Replace xxxx with the GitHub issue number)
 I have tried changing the groupby to GroupBy. Please check the hyperlink if it works.

### CODE REVIEW

File: pandas/core/groupby/groupby.py [hunk 1]
```diff
@@ -414,9 +414,9 @@ class providing the base-class of operations.
 
 See Also
 --------
-{klass}.groupby.apply : Apply function func group-wise
+{klass}.GroupBy.apply : Apply function func group-wise
```
Comment (mroeschke) [inline pandas/core/groupby/groupby.py:417]:
```suggestion
{klass}GroupBy.apply : Apply function func group-wise
```

File: pandas/core/groupby/groupby.py [hunk 2]
```diff
@@ -414,9 +414,9 @@ class providing the base-class of operations.
...
     and combine the results together.
-{klass}.groupby.transform : Transforms the Series on each group
+{klass}.GroupBy.transform : Transforms the Series on each group
```
Comment (mroeschke) [inline pandas/core/groupby/groupby.py:419]:
```suggestion
{klass}GroupBy.transform : Transforms the Series on each group
```

**Comment by doshi-kevin** [comment]
On it

**Comment by doshi-kevin** [comment]
Hey @mroeschke , I made the necessary changes. Could you please check and merge the PR . I really would like to contribute to such amazing projects like pandas.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @doshi-kevin

**Pull request merged**